src/js/load.js

```diff
@@ -40,13 +40,13 @@ export default function configureApplication(profileId, config) {
         config.analytics.registerEvents(controller);
 
     const mapcontrol = new MapControl(config);
-    mapcontrol.popup = new Popup(mapcontrol.map);
+    mapcontrol.popup = new Popup(config, mapcontrol.map);
```

Comment:
Currently, we're only interested in the `formatting` key of the config. 

What do you think about changing it so that we pass a formatting object into the Popup and LocationInfoBox.
It would reduce a lot of duplication. 

And we can provide a default right here. 
Which is a big thing. After discussing this with @adieyal the ticket did not mention a default but it is possible and very likely that there is no formatting key present in the config. And therefore we need to have a default here.

Reply:
in the future maybe we will need another key of the config, thats why I generally prefer to send the whole object. if you want I can change it though

Reply:
@adieyal how likely will it be that we will need other keys from the config? and how soon will this be?

If it is unlikely and far away I would change this to just using formatting for now and we can refactor this when we need to in the future. 
It will make the code easier to read and remove duplication.

Reply:
Are you talking about arbitrary keys or formatting specific? The rationale behind sending the config is that it is a really lightweight way of adding new features without affecting the backend. This allows experimenting while we are learning about the product's requirements. This means that we are likely to add arbitrarily keys on an ad hoc basis. An example that is emerging right now is letting the Profile Administrator decide whether point themes are open or closed by default. 

I wonder whether the easiest way to ensure that we have appropriate defaults is to use es6's object restructuring syntax. e.g.

```
defaultConfig = {
    key1: value1,
    key2: value2,
    ...
}

serverConfig = getConfigFromServer()
config = {...defaultConfig, ...serverConfig)
```
that way we can ensure that the config object has the right shape when we receive it from the server and avoid checks in the rest of the code.

Having said that, a formatting object is a good idea and the cost of refactoring is not high. In general, avoiding access object properties directly is a good idea so adding methods allows us to change business logic later on.

Reply:
I have a lot of ideas about this one. 
But since this must be shipped soon, let's go for the easy solution today and refactor this later.

@emre2038 can you change the code to just pass in the formatter key to the instances?

```js
let defaultFormattingConfig = {
    "decimal": "...",
    "percentage": "...",
    ...
}
let serverFormattingConfig = config.config;
let formattingConfig = { ...defaultFormattingConfig, ...serverFormattingConfig};
mapcontrol.popup = new Popup(formattingConfig, mapcontrol.map);
```

and update the code accordingly. 
thanks.

Reply:
changed as requested. @adieyal I added a default dict for formatting(variable name defaultFormattingConfig) in load.js is that ok? since last time I worked on load.js you've made so much changes :) let me know if you want me to move it somewhere else

Reply:
load.js will probably be refactored in future so I suppose it's a good a place as any. I would like to propose in a future refactoring that we don't pass the dictionary directly into Popup (or wherever else it is used) but rather to create a formatting class that will use the configuration. There are at three reasons to do this:

1. Passing a dictionary is dangerous if some code modifies it for whatever reason. The dictionary cannot enforce its own integrity whereas a class can implement setters and getters to control how it is used.
2. Separation of concerns. In future we may want to add additional functionality to the formatting class. Suppose for example we want to isolate the d3 format function so that it can be replaced with something else. It would be very costly to do this using a dictionary because d3 is used in the client classes (popup in this instance). Popup's job is presentation, not formatting. 
3. Testing is easier since you are injecting the service into the class which could be mocked with something else in future if needed.

It's quite cheap to do right now and I don't see any downsides.

Thoughts

Reply:
sorry, didn't notice the `utils/formatNumericalValue` but the point still holds, it should be passed into the constructor (or wherever appropriate) rather than being imported directly by the class that is using it.